### PR TITLE
feat(snap-keyring-bridge): sanitize redirect URL

### PR DESCRIPTION
**Pull request by ccharly**
Sanitize redirect URL when using the Snap keyring bridge.

### CODE REVIEW

**Comment by ccharly** [comment]
@metamaskbot publish-preview

**Comment by ccharly** [comment]
@metamaskbot publish-preview

**Comment by ccharly** [comment]
@metamaskbot publish-preview

**Comment by ccharly** [comment]
@metamaskbot publish-preview

File: packages/keyring-snap-bridge/package.json [hunk 1]
```diff
@@ -51,9 +52,6 @@
     "typedoc": "^0.25.13",
     "typescript": "~5.4.5"
   },
-  "peerDependencies": {
```
Comment (danroc) [inline packages/keyring-snap-bridge/package.json:54]: Why this change? I think we still use some `keyring-api` types in our public API 🤔

File: packages/keyring-snap-bridge/src/util.ts [hunk 2]
```diff
@@ -78,3 +78,16 @@ export function throwError(message: string): never {
...
 }
+
+/**
+ * Sanitize a URL.
+ *
+ * @param url - The URL to sanitize.
+ * @returns The new sanitized redirect URL.
+ */
+export function sanitizeUrl(url: string): string {
+  // For now, we only re-create the URL object which should take care of most of the sanitizing, like replacing
```
Comment (danroc) [inline packages/keyring-snap-bridge/src/util.ts:89]: Why "most" sanitizing? Do you have other examples where it's not sanitized correctly?

File: packages/keyring-snap-bridge/package.json (hunk 1 above)
Comment (ccharly) [reply packages/keyring-snap-bridge/package.json:54]: That was mainly to make the preview builds work (so I can use them in the extension afterward)

But yes, I'll definitely revert those changes before merging 👍

File: packages/keyring-snap-bridge/src/util.ts (hunk 2 above)
Comment (ccharly) [reply packages/keyring-snap-bridge/src/util.ts:89]: Good question, the wording is probably not the best here.  Maybe something like:
```ts
// We use a `URL` object to detect any badly formatted URL and to normalize/sanitize them.
```

WDYT?

**Comment by ccharly** [comment]
@metamaskbot publish-preview

**Comment by ccharly** [comment]
@metamaskbot publish-preview

**Comment by ccharly** [comment]
@metamaskbot publish-preview

File: packages/keyring-snap-bridge/package.json (hunk 1 above)
Comment (ccharly) [reply packages/keyring-snap-bridge/package.json]: I did revert that back now!

**Review by danroc — APPROVED** [review]
